### ISSUE TITLE
[Bug]: eu.prod.push.monal-im.org presents invalid certificate

### ISSUE THREAD

**Issue by svenschwermer** [monal-im/Monal#1203]
### What Monal Release channel are you using?

AppStore

### Monal version

N/A

### Used XMPP server (domain)

N/A

### Which XMPP-Server software are you using?

None

### How many accounts are you using in Monal?

1

### What happened?

The HTTPS certificate eu.prod.push.monal-im.org presents is invalid (it only covers the hostname `downloads.monal-im.org`):

```
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            03:d8:89:8d:f0:cd:63:61:ad:32:95:2f:4d:b5:87:c7:77:4b
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: C=US, O=Let's Encrypt, CN=R11
        Validity
            Not Before: Jun 23 02:25:28 2024 GMT
            Not After : Sep 21 02:25:27 2024 GMT
        Subject: CN=downloads.monal-im.org
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                Public-Key: (4096 bit)
                Modulus:
                    00:ab:bf:fa:d3:8a:8b:25:35:3d:41:7b:23:01:1e:
                    44:fa:65:4b:91:2d:07:d6:52:92:bb:68:39:82:97:
                    ed:20:10:50:ff:24:db:bd:5e:bc:d1:4d:9c:fe:10:
                    e6:70:ad:b5:c7:bc:dc:aa:fe:fc:24:e4:5b:2f:0f:
                    3b:f3:78:30:69:b8:16:eb:41:67:87:5a:de:9e:49:
                    98:74:22:f9:12:6f:54:9b:1f:35:0b:92:ef:5d:08:
                    9c:01:93:52:cd:fc:da:f1:62:8a:1f:7a:c4:2f:f3:
                    dc:6e:c8:d4:51:43:db:b7:57:20:b5:b5:c8:1d:00:
                    ea:5d:f6:30:de:9d:b2:b3:77:74:16:25:ff:d3:99:
                    1b:5c:f1:db:d1:04:fa:dc:21:9f:20:c1:e8:05:7b:
                    e2:cd:ca:51:e5:45:42:51:2a:b8:31:4e:9a:7d:7f:
                    fc:08:ed:90:cb:35:dd:ed:91:96:ce:3d:7f:4b:64:
                    13:de:59:1e:bc:e0:ef:27:1b:26:eb:72:1b:cd:f0:
                    ae:c2:82:5a:21:c7:22:49:d3:ab:36:2b:a2:91:80:
                    86:43:7c:45:dc:68:de:a0:25:c0:86:da:0e:46:3c:
                    1f:40:27:89:3b:a9:d3:07:51:b9:64:f3:3d:bf:e5:
                    7d:9b:84:f3:6b:de:54:7a:8e:a0:3a:9d:e1:5b:03:
                    31:fc:4d:df:3c:60:e9:89:6b:c1:27:99:78:8d:3a:
                    2d:fc:8b:08:61:31:35:73:57:bb:7b:4e:90:d1:4a:
                    80:62:38:a7:a1:5f:a1:2f:10:54:52:e7:d9:36:1e:
                    81:28:f1:6d:b9:98:8b:92:7b:3c:33:5e:5a:18:1b:
                    2e:cf:83:0f:b3:7f:f2:33:49:e9:64:59:6b:31:1b:
                    b0:26:64:f5:df:90:e5:38:98:a7:02:31:4f:3b:8a:
                    e9:6a:1a:15:0d:d6:75:50:6f:79:cb:0e:7a:74:7d:
                    d1:0f:a2:bf:9e:e9:72:1b:52:41:dd:f0:26:fc:07:
                    e2:7b:c3:5e:46:8a:69:f7:59:2a:8f:ba:6d:38:b8:
                    b1:b4:33:ec:39:96:09:4d:28:8f:2a:e8:1a:28:57:
                    99:51:68:13:1a:62:70:cc:40:25:c7:3f:3e:3f:96:
                    f6:b4:98:cc:e0:33:0c:bf:d7:04:ec:89:05:17:79:
                    77:46:09:4b:ea:ea:2d:0f:43:bb:06:7c:c5:17:7b:
                    c9:b9:e7:e3:9a:f1:72:5d:52:ed:7a:bd:ea:ef:cf:
                    00:21:3a:d8:0f:86:eb:f6:e6:cb:41:4d:36:4f:98:
                    e4:e0:1d:0e:df:aa:cb:12:dd:96:f3:2f:6c:49:26:
                    81:f7:e0:db:08:36:c6:56:3f:52:e7:b0:a6:14:8f:
                    d1:f9:51
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Key Usage: critical
                Digital Signature, Key Encipherment
            X509v3 Extended Key Usage: 
                TLS Web Server Authentication, TLS Web Client Authentication
            X509v3 Basic Constraints: critical
                CA:FALSE
            X509v3 Subject Key Identifier: 
                78:D4:85:1B:38:04:C2:F3:AA:D2:78:63:60:29:93:A8:AD:C0:FC:F0
            X509v3 Authority Key Identifier: 
                C5:CF:46:A4:EA:F4:C3:C0:7A:6C:95:C4:2D:B0:5E:92:2F:26:E3:B9
            Authority Information Access: 
                OCSP - URI:http://r11.o.lencr.org
                CA Issuers - URI:http://r11.i.lencr.org/
            X509v3 Subject Alternative Name: 
                DNS:downloads.monal-im.org
            X509v3 Certificate Policies: 
                Policy: 2.23.140.1.2.1
            CT Precertificate SCTs: 
                Signed Certificate Timestamp:
                    Version   : v1 (0x0)
                    Log ID    : 19:98:10:71:09:F0:D6:52:2E:30:80:D2:9E:3F:64:BB:
                                83:6E:28:CC:F9:0F:52:8E:EE:DF:CE:4A:3F:16:B4:CA
                    Timestamp : Jun 23 03:25:28.384 2024 GMT
                    Extensions: none
                    Signature : ecdsa-with-SHA256
                                30:45:02:21:00:B7:2B:BF:74:DE:00:36:FD:D7:6E:E8:
                                8B:AF:D6:38:BA:C6:45:CE:5B:01:2A:50:11:CB:03:93:
                                4A:E0:EC:A2:8F:02:20:76:46:8C:99:B2:68:B0:CE:88:
                                A3:2F:B3:9E:B6:AD:44:21:02:DE:93:BA:3D:10:D4:E1:
                                05:5A:19:3A:E7:E7:50
                Signed Certificate Timestamp:
                    Version   : v1 (0x0)
                    Log ID    : 48:B0:E3:6B:DA:A6:47:34:0F:E5:6A:02:FA:9D:30:EB:
                                1C:52:01:CB:56:DD:2C:81:D9:BB:BF:AB:39:D8:84:73
                    Timestamp : Jun 23 03:25:28.245 2024 GMT
                    Extensions: none
                    Signature : ecdsa-with-SHA256
                                30:45:02:20:44:45:2B:36:DE:00:4F:A9:94:71:26:18:
                                8D:F2:D9:20:87:00:28:CB:01:55:F7:BE:6F:B0:74:D0:
                                92:84:1F:BF:02:21:00:94:85:EB:6D:1E:FD:D9:C0:14:
                                16:EA:E9:9B:11:01:CD:1A:03:C4:D4:81:1C:88:B9:71:
                                0D:79:09:C6:E3:F5:6E
    Signature Algorithm: sha256WithRSAEncryption
    Signature Value:
        16:41:82:6c:a1:f2:80:36:6a:0b:f2:e7:30:9a:b9:88:77:d6:
        ef:70:37:e6:ec:55:e0:74:cd:1f:04:51:d2:62:41:7b:a6:54:
        9e:12:95:c0:3e:35:23:b2:41:83:6d:bb:55:d2:28:86:42:d4:
        ec:20:ee:16:0f:9c:a5:26:f2:30:a0:a2:b3:53:39:a4:1c:04:
        65:e5:4a:a1:f0:4b:0b:62:8c:ff:b1:96:7d:4e:84:e9:0b:2d:
        9a:88:35:09:f1:7d:f9:2d:86:18:dc:a7:91:f6:52:50:3f:21:
        e0:75:e7:d1:5c:50:80:ba:86:d0:2f:9a:5d:c8:d3:f4:eb:f0:
        87:8f:4c:49:eb:30:84:45:db:fd:c2:e1:ce:70:4f:3a:da:cc:
        fe:3c:bc:4a:a3:47:5e:10:8f:74:8f:a0:3a:d9:5b:94:c1:d1:
        82:35:b8:09:ec:f2:92:37:25:a8:60:9d:6c:8a:14:22:8c:15:
        e3:49:8b:8c:2f:f8:ff:02:77:b6:0a:c1:5d:dc:c4:60:e5:d0:
        f1:5c:d2:11:df:fa:b2:36:c8:8c:2e:67:65:57:9a:a8:84:9c:
        cc:6f:da:80:ea:06:2b:2f:15:ec:00:ef:8d:81:6b:4d:20:89:
        cc:59:28:10:92:42:b0:02:fc:8d:21:f1:7e:2e:58:2a:7b:12:
        0b:f6:62:36
```

This makes prosody complain when trying to send a push notification (rightfully so).

### FAQ

- [X] I have [checked the FAQ](https://github.com/monal-im/Monal/wiki/FAQ---Frequently-Asked-Questions) for my issue

### Considerations for XMPP users

- [X] I have checked if my issue can be solved with [Considerations for XMPP users](https://github.com/monal-im/Monal/wiki/Considerations-for-XMPP-users) and [Considerations for XMPP server admins](https://github.com/monal-im/Monal/wiki/Considerations-for-XMPP-server-admins)

### XEP-Check

- [X] I have checked that at least XEP-198, XEP-0280, XEP-0352, XEP-0357, XEP-0313 and XEP-0163 are activated on my server and shown as 'green' under Settings --> Account--> (i) in advanced settings

### Notifications-Menu

- [X] I have checked that all checkmarks are present under Settings --> Notifications

**Comment by FriedrichAltheide** [comment]
Both of our two push servers have valid certificates. Hence, I am quite puzzled that ejabberd does not use the right one. (Yet, I am aware of an annoying ejabberd bug where the server - apart from the first loaded cert - can not remember the other loaded certs. And at some point in time these certs are known again...).

Could you please have a look if you are connecting via 5270 or 5269 and which of our servers was used (IP).

Note, I have have left ejabberd untouched for now

**Comment by FriedrichAltheide** [comment]
Based on how many servers are able to connect, you should have a look at your XMPP server. (e.g., if you server is using SNI as expected)

**Comment by svenschwermer** [comment]
Thanks for your quick reply. I got the above certificate via HTTPS (TCP port 443). However, I checked (using tcpdump) what network connection prosody actually makes: it dials port 5270. With SNI, I get a valid certificate. I will have to dig into more verbose prosody logs to find what exactly fails to verify... If necessary, I'll re-open.

edit: This is embarrassing: I had forgotten to install the root certificates in the prosody container image :blush: After installing those, everything works as expected.
